addons/common/fnc_compatibleMagazines.sqf

```diff
@@ -0,0 +1,54 @@
+#include "script_component.hpp"
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_compatibleMagazines
+
+Description:
+    Retrievs a list of magazines that are compatible with a weapon.
+
+Parameters:
+    _weapon           - Weapon configName or config
+
+Example:
+    (begin example)
+    _mags = ["arifle_MX_SW_F"] call CBA_fnc_compatibleMagazines
+    (end)
+
+Returns:
+    Array of magazine classnames <ARRAY>
+
+Author:
+    PabstMirror, based on code from Dedmen
+---------------------------------------------------------------------------- */
+SCRIPT(compatibleMagazines);
+
+params [["_weapon", "", ["", configNull]]];
+
+if (_weapon isEqualType "") then {
+    _weapon = configFile >> "CfgWeapons" >> _weapon;
+};
+
+private _cacheKey = str _weapon;
+if (_cacheKey == "") exitWith { ERROR_1("Weapon Does Not Exist %1",_this); [] };
+
+if (isNil QGVAR(magNamespace)) then { GVAR(magNamespace) = call CBA_fnc_createNamespace; };
+
+private _returnMags = GVAR(magNamespace) getVariable _cacheKey;
+
+if (isNil "_returnMags") then {
```

Comment:
this could be replaced with early return
`if (!isNil "_returnMags") exitWith {_returnMags};`

Reply:
Don't forget to copy the returned array, otherwise someone will mess up the cache by editing it by reference.